### PR TITLE
[Merge] call prepareBeaconProposer on every new epoch

### LINKED ISSUE #4659: [Merge] call `prepare_beacon_proposer` in validator client

### PR DESCRIPTION
**Pull request by tbenr**
## PR Description
call prepareBeaconProposer on every new epoch, let beacon node know which are the potential proposers that might need to produce a block with `ExecutionPayload`

moves `--Xee-fee-recipient-address` under validator config `--Xvalidators-fee-recipient-address`

## Fixed Issue(s)
fixes #4659

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/BeaconProposerPreparer.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    this.validatorIndexProvider = validatorIndexProvider;
+    this.validators = validators;
+    this.feeRecipient = feeRecipient.orElse(null);
+    this.spec = spec;
+  }
+
+  @Override
+  public void onSlot(UInt64 slot) {
+    if (spec.atSlot(slot).getExecutionPayloadUtil().isPresent()
+        && slot.mod(spec.getSlotsPerEpoch(slot)).isZero()) {
```
Comment (tbenr) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/BeaconProposerPreparer.java]: just asking my self if it is safe for the case of being the proposer of the first slot in Merge fork. We might start calling this from the prior epoch?

Comment (ajsutton) [reply validator/client/src/main/java/tech/pegasys/teku/validator/client/BeaconProposerPreparer.java]: I don't see any reason this shouldn't be called prior to the merge - there's no point if the merge fork isn't enabled at all but having the beacon node notified ahead of the merge happening doesn't hurt and means the beacon node is definitely ready when the merge fork kicks in.  So I think I'd actually remove the `getExecutionPayloadUtil().isPresent()` check here.

The beacon node just wouldn't do anything with the information about which validators are running until the merge epoch actually kicks in (and then probably only if terminal total difficulty is reached).

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/BeaconProposerPreparer.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+          .getValidatorIndices(validators.getPublicKeys())
+          .thenApply(
+              integers ->
+                  integers.stream()
+                      .map(
+                          index ->
+                              new BeaconPreparableProposer(UInt64.valueOf(index), feeRecipient))
+                      .collect(Collectors.toList()))
+          .thenAccept(validatorApiChannel::prepareBeaconProposer)
+          .reportExceptions();
```
Comment (ajsutton) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/BeaconProposerPreparer.java]: `.reportExceptions()` probably isn't the right thing here.  That reports everything via the default exception handler which asks users to report it as a bug.

I think we want to report errors via something in `ValidatorLogger` similar to `ValidatorLogger.dutyFailed`.

**Review by ajsutton — APPROVED** [review]
LGTM generally just a couple of tweaks.

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/BeaconProposerPreparer.java (hunk 1 above)
Comment (tbenr) [reply validator/client/src/main/java/tech/pegasys/teku/validator/client/BeaconProposerPreparer.java]: oky.
my initial concern was a non-problem: forgot to consider that TTD needs to be reached before this data become necessary.

**Pull request merged**